src/components/PersonLink.tsx

```diff
@@ -0,0 +1,84 @@
+import { Link, useLocation, useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+import classNames from 'classnames';
+import { GetPersonDates } from '../services/getPersonDates';
+
+type Props = {
+  person: Person;
+};
+
+export const PersonLink: React.FC<Props> = ({ person }) => {
+  const { name, sex, born, died, slug, motherName, fatherName } = person;
+
+  const MOTHER_NAME = motherName;
+  const FATHER_NAME = fatherName;
+
+  const { pathname } = useLocation();
+  const [searchParams] = useSearchParams();
+
+  const onlySlug = pathname.split('/').slice(2).join();
```

Comment:
The `onlySlug` variable is constructed by splitting the pathname and joining it back. This might not work correctly if the pathname structure changes. Consider using a more reliable method to extract the slug, such as using URL parameters or a regex.